lib/src/components/NxVulnerabilityIndicator/NxVulnerabilityIndicator.tsx

```diff
@@ -0,0 +1,63 @@
+/*
+ * Copyright (c) 2019-present Sonatype, Inc.
+ * This program and the accompanying materials are made available under
+ * the terms of the Eclipse Public License 2.0 which accompanies this
+ * distribution and is available at https://www.eclipse.org/legal/epl-2.0/.
+ */
+import React, { FunctionComponent, ReactElement } from 'react';
+import NxTooltip from '../NxTooltip/NxTooltip';
+import NxFontAwesomeIcon from '../NxFontAwesomeIcon/NxFontAwesomeIcon';
+import { faSquare } from '@fortawesome/free-solid-svg-icons';
+
+import {
+  vulnerabilitySeverityScoreRangeMap,
+  getVulnerabilitySeverityRating,
+  VulnerabilitySeverityRating
+} from '../../util/vulnerabilitySeverityRatings';
+import { Props, propTypes } from './types';
+export { Props } from './types';
+
+import './NxVulnerabilityIndicator.scss';
+import classnames from 'classnames';
+
+const capitalize = (string: string) => string.charAt(0).toUpperCase() + string.slice(1);
+
+const NxVulnerabilityIndicator: FunctionComponent<Props> =
+  function NxVulnerabilityIndicator(props): ReactElement<Props> {
+    const {
+          title,
+          severityRating,
+          score,
+          className: classNameProp,
+          presentational
+        } = props,
+        severity: VulnerabilitySeverityRating = severityRating
+          ? severityRating
+          : score !== undefined
+            ? getVulnerabilitySeverityRating(score) || 'none'
+            : 'none',
+        className = classnames(
+            classNameProp,
+            `nx-vulnerability-indicator nx-vulnerability-indicator--${severity} nx-icon--colorful`
+        ),
+        label = `vulnerability rating ${severity}`;
+
+    const icon = (
+      <NxFontAwesomeIcon aria-label={presentational ? undefined : label}
+                         className={className}
+                         icon={faSquare}
+                         role={presentational ? 'presentation' : 'graphics-symbol img'} />
+    );
+
+    return presentational ? icon : (
+      <NxTooltip title={
+        title || `Vulnerability: ${capitalize(severity)} (${vulnerabilitySeverityScoreRangeMap[severity]})`}
```

Comment:
This is too complex and long to put inside the JSX like this.  Extract it to a local variable.

Reply:
Fixed formatting here: [71c9993](https://github.com/sonatype/sonatype-react-shared-components/pull/1199/commits/71c9993f1af37895cae88b020c99f9047badf26a)